discovery/service_test.go

```diff
@@ -383,6 +411,11 @@ func createTestCtx(startHeight uint32) (*testCtx, func(), error) {
 	notifier := newMockNotifier()
 	router := newMockRouter(startHeight)
 
+	db, cleanUpDb, err := makeTestDB()
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
In order to ensure the db is _always_ cleaned up, a `defer cleanUpDb()` should be inserted here.